Sources/NostrSDK/EventCreating.swift

```diff
@@ -173,7 +174,28 @@ public extension EventCreating {
 
         return try ReactionEvent(content: content, tags: tags, signedBy: keypair)
     }
-    
+
+    /// Creates a ``ReactionEvent`` (kind 7) in response to a different ``NostrEvent`` and signs it with the provided ``Keypair``.
+    /// - Parameters:
+    ///   - customEmoji: The custom emoji to emojify with if the matching shortcode is found in the content field.
+    ///   - reactedEvent: The NostrEvent being reacted to.
+    ///   - keypair: The Keypair to sign with.
+    /// - Returns: The signed ``ReactionEvent``.
+    ///
+    /// See [NIP-25 - Reactions](https://github.com/nostr-protocol/nips/blob/master/25.md)
+    func reaction(withCustomEmoji customEmoji: CustomEmoji, reactedEvent: NostrEvent, signedBy keypair: Keypair) throws -> ReactionEvent {
```

Comment:
Do we need a separate function for this? Or could we add it to the existing one and make both the `content` and `customEmoji` optional with a default value of nil? I don't feel strongly either way.

Reply:
I think it's better to have a separate function. It's an XOR situation -- I want to avoid having the ability to define `content` that doesn't use the custom emoji shortcode from the emoji tag because that's out of spec.

Reply:
Yeah, it matches well if it's an XOR thing. I'm not familiar enough with these custom emojis.